### PR TITLE
API Extractor Doc fixes

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
API Extractor will be used to generate the reference documentation for firestore-exp. It currently throws some validation errors, and this fixes them by doing the following:

- Use `@returns` instead of `@return`
- Add "-" for `@param`
- HTML escape special characters
- Fix `@link` tags
- Add `@hideconstructor` for non-public (internal) constructors

I don't think this needs a thorough review, but a spot check would be nice.

### CODE REVIEW

File: packages/firestore/src/local/index_manager.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ import { PersistencePromise } from './persistence_promise';
 /**
  * Represents a set of indexes that are used to execute queries efficiently.
  *
- * Currently the only index is a [collection id] => [parent path] index, used
+ * Currently the only index is a [collection id] =&gt; [parent path] index, used
```
Comment (Feiyang1) [inline packages/firestore/src/local/index_manager.ts:25]: Why do we need to escape it? Does it not work in markdown?

**Review by dconeybe — APPROVED** [review]
LGTM

File: packages/firestore/src/local/index_manager.ts (hunk 1 above)
Comment (schmidt-sebastian) [reply packages/firestore/src/local/index_manager.ts:25]: API Extractor complains about all these HTML-like characters (at least with default settings).

File: packages/firestore/exp/src/api/snapshot.ts [hunk 2]
```diff
@@ -88,14 +88,21 @@ export interface FirestoreDataConverter<T> {
    * `toFirestore()` must be defined with `Partial<T>`.
    */
   toFirestore(modelObject: T): DocumentData;
+
+  /**
+   * Called by the Firestore SDK to convert a custom model object of type `T`
+   * into a plain Javascript object (suitable for writing directly to the
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply packages/firestore/exp/src/api/snapshot.ts]: Done

**Pull request merged**